code-challenges.js

```diff
@@ -9,38 +9,111 @@
 // --------------------1) Create a function that takes in a number and determines if a temperature is below boiling point, at boiling point or above boiling point. Boiling point is 212 degrees Fahrenheit.
 // Use the test variables provided below. Expected output: "35 is below boiling point", "350 is above boiling point", "212 is at boiling point"
 
+// I want to create a function expression that will take in the temp variables.
+//I should use some if/else statements to determine if the temp is boiling or below boiling
+//I want to return a string
+//potentially check if a given input is valid 
+
+
 var temp1 = 35
 var temp2 = 350
 var temp3 = 212
 
+const temperatureChecker = (temp) => {
+   if(typeof temp !== "number") return "Please enter a valid number";
+   
+   if(temp === 212){
+      return `${temp} is boiling point`; 
+   } else if(temp < 212) {
+     return `${temp} is below boiling point`;  
+   } else {
+    return `${temp} is above boiling point`;    
+   }
+}
+
+//console.log(temperatureChecker(temp3));
 
 
 
 
 // --------------------2) Create a function that takes in an array of numbers and returns an array with each number multiplied by 5.
 // Use the test variable provided below. Expected output: [15, 35, 0, 30, -45]
 
+//I want to create a function expression that takes in an array and return the array multiplied by 5
+//Create a empty return array;
+//perform a for loop through all the elements within the array 
+//take each element multiply it by 5 and push it into the empty array;
+//return the array.
+
 var myNumbers1 = [3, 7, 0, 6, -9]
 
+const multiply5 = (myNumbers1) => {
+   let newArray = [];
+   
+   for (let i = 0; i < myNumbers1.length; ++i){
+      newArray.push(myNumbers1[i] * 5);
+   }
+   return newArray;
+}
+
+//console.log(multiply5(myNumbers1));
```

Comment:
The purpose of a function is to create a generic and reusable code. So what gets passed into the function should be a parameter and not the actual variable.